magicctapipe/scripts/lst1_magic/semi_automatic_scripts/check_MAGIC_runs.py

```diff
@@ -0,0 +1,134 @@
+
+def fix_lists_and_convert(cell):
+    # Remove brackets to avoid double lists and split on ']['
+    parts = cell.replace('][', ',').strip('[]').split(',')
+    return list(dict.fromkeys(int(item) for item in parts))
+
+def table_magic_runs(df, date_min, date_max):
+    df_selected_data = df.iloc[:, [2, 1, 25]]
+    df_selected_data.columns = ['DATE','source', 'MAGIC_runs']
+    grouped_data = df_selected_data.groupby(['DATE', 'source'])
+    result_table = []
+
+    for (date, source), group in grouped_data:
+        if (date>=date_min and date<=date_max):  
+            runs_combined = group['MAGIC_runs'].sum()
+    
+            result_table.append({
+                'DATE': date,
+                'source': source,
+                'MAGIC runs': runs_combined
+            })
+        
+    result = pd.DataFrame(result_table)
+    result['MAGIC runs'] = result['MAGIC runs'].apply(fix_lists_and_convert)
+    return(result)
+
+def existing_files( tel_id, date, source, magic_runs ):
+
+    magic_runs = str(magic_runs)  
+    date_obj = datetime.strptime(date, '%Y%m%d')
+    date_obj += timedelta(days=1)
+    new_date = datetime.strftime(date_obj, '%Y%m%d')
+    YYYY = new_date[:4]
+    MM = new_date[4:6]
+    DD = new_date[6:8]
+    Y = f"_Y_"
+    
+    path = f"/fefs/onsite/common/MAGIC/data/M{tel_id}/event/Calibrated/{YYYY}/{MM}/{DD}" 
+    
+    if os.path.exists(path):
+        files = os.listdir(path)
+        count_with_source = 0  
+        count_with_run_id = 0
+            # Counter for files that include the source.  
+            # Counter for files that include the run_id.
+        for filename in files:
+            if date and source and Y in filename:
+                count_with_source += 1
+                if magic_runs in filename:
+                    count_with_run_id += 1
+        if count_with_source != 0 and count_with_run_id != 0:
+            print(f"{date}\t{source}\t{magic_runs}\t{count_with_run_id}")
+                    
+def missing_files( tel_id, date, source, magic_runs ):
+    
+    for runs in magic_runs:
+        run = str(runs)
+    
+    date_obj = datetime.strptime(date, '%Y%m%d')
+    date_obj += timedelta(days=1)
+    new_date = datetime.strftime(date_obj, '%Y%m%d')
+    YYYY = new_date[:4]
+    MM = new_date[4:6]
+    DD = new_date[6:8]
+    Y = f"_Y_"
+    
+    path = f"/fefs/onsite/common/MAGIC/data/M{tel_id}/event/Calibrated/{YYYY}/{MM}/{DD}" 
+    
+    if os.path.exists(path):
+        files = os.listdir(path)
+        count_with_source = 0  
+        count_with_run_id = 0
+        # Counter for files that include the source. We want to check if any file with the source was found. 
+        # Counter for files that include the run_id. We want to check if any file with the run_id was found. 
+        for filename in files:
+            if date and source and Y in filename:
+                count_with_source += 1
+                for runs in magic_runs:
+                   # run = str(runs)
```

Comment:
here you do something strange, you loop over all the runs, but every time you check only the last one (see my comment to line 70)

Reply:
Yes, I see... Thanks! I fixed it and I will add all the corrections soon